src/mod/server.mod/servmsg.c

```diff
@@ -1132,6 +1141,115 @@ static int got311(char *from, char *msg)
   return 0;
 }
 
+static int gotauthenticate(char *from, char *msg)
+{
+  char src[256] = ""; // FIXME: size
+  char *s;
+  size_t slen;
+  unsigned char dst[256] = ""; // FIXME: size
+  size_t olen;
+  unsigned char dst2[256] = ""; // FIXME: size
+  unsigned int olen2;
+#ifdef HAVE_OPENSSL_SSL_H
+  FILE *fp;
+  EC_KEY *eckey;
+  EVP_PKEY *privateKey;
+#endif
+
+  putlog(LOG_SERV, "*", "SASL: got AUTHENTICATE %s", msg);
+  if (msg[0] == '+') {
+    s = src;
+    /* Don't use snprintf due to \0 inside */
+    if (!strcmp(sasl_mechanism, "PLAIN")) {
```

Comment:
We switched this to digits in the config, should be 0, 1, 2?

Reply:
fixed. the code still treated sasl_mechanism as string not enum.